kubernetes/samples/scripts/domain-lifecycle/helper.sh

```diff
@@ -111,14 +111,20 @@ function createServerStartPolicyPatch {
   local policy=$3
   local __result=$4
   local currentStartPolicy=""
+  local serverStartPolicyPatch=""
 
   # Get server start policy for this server
   getServerPolicy "${domainJson}" "${serverName}" currentStartPolicy
-  if [ -z "${currentStartPolicy}" ]; then
+  managedServers=$(echo ${domainJson} | jq -cr '(.spec.managedServers)')
+  if [[ -z "${currentStartPolicy}" && "${managedServers}" == "null" ]]; then
     # Server start policy doesn't exist, add a new policy
     addPolicyCmd=".[.| length] |= . + {\"serverName\":\"${serverName}\", \
       \"serverStartPolicy\":\"${policy}\"}"
     serverStartPolicyPatch=$(echo ${domainJson} | jq .spec.managedServers | jq -c "${addPolicyCmd}")
+  elif [ "${managedServers}" != "null" ]; then
+    extractSpecCmd="(.spec.managedServers)"
+    mapCmd=". |= (map(.serverName) | index (\"${serverName}\")) as \$idx | if \$idx then .[\$idx][\"serverStartPolicy\"] = \"${policy}\" else .+  [{serverName: \"${serverName}\" , serverStartPolicy: \"${policy}\"}] end"
```

Comment:
This is a long line. Something like the following would be easier to read:
```
mapCmd="\
  . |= (map(.serverName) | index (\"${serverName}\")) as \$idx | \
  if \$idx \
  then .[\$idx][\"serverStartPolicy\"] = \"${policy}\" \
  else .+  [{serverName: \"${serverName}\" , serverStartPolicy: \"${policy}\"}] end "
```
(I may have a syntax error in above - it's just intended as an example.)